.github/workflows/build-user-config.yml

```diff
@@ -23,6 +23,21 @@ on:
         default: "firmware"
         required: false
         type: string
+      release_enable:
+        description: "Enable publishing of artifacts as a release"
+        default: true
+        required: false
+        type: boolean
+      release_name:
+        description: "Release name"
+        default: "ZMK Firmware"
+        required: false
+        type: string
+      release_tag:
+        description: "Release tag"
+        default: "latest"
```

Comment:
It seems like you're just constantly re-tagging whatever commit is getting built as "latest" every time the workflow is run? Rewriting Git history is not great, and probably not something we want to do by default (or even support at all). What happens if you push two commits in quick succession, and "latest" gets moved in the middle of one of the builds?

Typically, you would use action-gh-release by setting up a separate workflow that triggers when a tag is added, or by adding it to the existing workflow and gating the release steps with
```yaml
  if: startsWith(github.ref, 'refs/tags/')
```

so that it creates a new release for every tag you add. Then, a release is only created when you explicitly ask for one by creating a new tag, and since each new tag has a different name, all your releases stick around and reference the specific commits from which they were built.

Reply:
The purpose of this addition is to simply publish _only_ the latest artifact as a release with a single tag. Users can continue to find older artifacts in the Actions tab.

It is _not_ meant to publish new tags/versions in the conventional manner. Deleting the previous release ensures that latest release time matches the artifact content upload time to avoid ambiguity.

> Rewriting Git history is not great, and probably not something we want to do by default (or even support at all). 

If these workflow addition are not desired from a maintainer point of view, PR will be withdrawn.

Reply:
I think having a way to automatically generate releases could be useful, but if we added that, I would prefer that we use GitHub releases as they're intended to be used.

Reply:
> I think having a way to automatically generate releases could be useful, but if we added that, I would prefer that we use GitHub releases as they're intended to be used.

Thanks for the feedback.